src/number_format.js

```diff
@@ -547,8 +547,10 @@ class NumberFormat extends React.Component {
     const diff = deletedIndex !== -1 ? lastValueParts[1].substring(0, deletedIndex) : '';
     const end = start + diff.length;
 
+    const isDeleteAll = lastValueParts[1].length === deletedIndex;
```

Comment:
Wouldn't it will mark isDeleteAll for the case where all text after caretPos is selected. but there is text before caret position